libraries/WiFi/src/WiFiSTA.h

```diff
@@ -108,12 +108,20 @@ class WiFiSTAClass
 
     static void _setStatus(wl_status_t status);
     
+    void setBeginTimeout(unsigned long timeout) {
+        beginTimeout = timeout;
+    }
+    void setBeginAsync() {
+        beginTimeout = 0;
+    }
+
 protected:
     static bool _useStaticIp;
     static bool _autoReconnect;
     static wifi_auth_mode_t _minSecurity;
     static wifi_scan_method_t _scanMethod;
     static wifi_sort_method_t _sortMethod;
+    unsigned long beginTimeout = 60000;
```

Comment:
1 minute timeout is absolutely unusable.
10% of it might be something to consider, but a default timeout of 60 seconds is hardly every usable in the world of microcontrollers.

As was already mentioned, a default of 0 would be at least non-breaking, but then you explicitly need to set the timeout and if you would change your code to do so, then you probably already have a similar loop right after the call to `WiFi.begin()`